### PR TITLE
Add pricing calculator to pricing page

### PR DESCRIPTION
**Pull request by kimberleyamackenzie**
## Description
Add a pricing calculator to the pricing page - this started as a hackathon project, and @meagancojocar has reviewed and also run past the Sales team.  Allows inputs for resource count, utilization, and deployment minutes and shows outputs of costs for monthly and daily usage.  Also adds a quick button to the calculator at the top of the page.  See some screenshots below for responsiveness.
Screenshot 2023-11-27 at 2 48 24 PM
Screenshot 2023-11-27 at 2 48 43 PM
Screenshot 2023-11-27 at 2 48 56 PM
Screenshot 2023-11-27 at 2 50 54 PM
Screenshot 2023-11-27 at 2 51 09 PM
Screenshot 2023-11-27 at 2 51 20 PM
Screenshot 2023-11-27 at 2 54 44 PM

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

**Comment by meagancojocar** [comment]
We decided to ship this! Some changes to make:

- $20k per year cap, which is $1.7k a month, which is 5000 resources. When this amount is reached we want to change the $ amount to say "Contact sales for bulk discounts".
- Remove Enterprise. The calculator will just be for Team customers only.

File: themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+            }
+        }
+    }
+}
+
+.range {
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    width: 140px;
+    // margin: auto;
```
Comment (kimberleyamackenzie) [inline themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss:267]:
```suggestion
```

**Review by zchase — CHANGES_REQUESTED** [review]
We will want this run through our A/B testing tool. I'll make some time tomorrow to set it up.

File: themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss [hunk 2]
```diff
@@ -0,0 +1,348 @@
+@import "../../../../src/scss/colors";
+.calculator {
+    margin-right: 36px;
+    margin-left: 36px;
```
Comment (cnunciato) [inline themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss:4]: Would strongly prefer Tailwind classes for these.

Comment (kimberleyamackenzie) [reply themes/default/theme/stencil/src/components/pulumi-pricing-calculator/pulumi-pricing-calculator.scss:4]: i based this off other stencil components where we have custom styles + i didn't see any usage of tailwind in any of those cases, so i made the assumption that it was intentional/tailwind + stencil didn't play well together.  but if that's not the case i can go back and tailwind where possible.

**Review by zchase — APPROVED** [review]
All good for the A/B test!

**Review by meagancojocar — APPROVED** [review]
Good to merge!

**Comment by kimberleyamackenzie** [comment]
Cleaned up a couple items after syncing with @anita-trimbur - see screenshots below.  @meagancojocar I also found an error in math where I was not properly calculating the deployment minutes per month, so I fixed that too.

- [ ] Fixed bad math 😅 
- [ ] Decreased spacing next to calculator link at top
- [ ] Increased spacing below calculator header
- [ ] Gave the outputs a little more breathing room at below 1280
- [ ] Added "estimated deployment minutes" item in outputs to make it clear that the calculator was still reacting when that input was changed but remained below the free minutes
- [ ] Switched up the treatment for contact us to include link to sales form + blurred output

Screenshot 2023-11-28 at 11 52 49 AM
Screenshot 2023-11-28 at 11 57 07 AM


Screenshot 2023-11-28 at 11 52 42 AM

**Review by anita-trimbur — APPROVED** [review]

**Pull request merged**